core/domain/draft_upgrade_services.py

```diff
@@ -88,6 +90,135 @@ def try_upgrading_draft_to_exp_version(
+                  change.property_name ==
+                  exp_domain.STATE_PROPERTY_WRITTEN_TRANSLATIONS):
+                converted_new_value = (
+                    state_domain.WrittenTranslations.
+                    convert_html_in_written_translations(
+                        change.new_value,
+                        html_validation_service.
+                        add_math_content_to_math_rte_components))
+                draft_change_list[i] = exp_domain.ExplorationChange({
+                    'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                    'property_name': (
+                        exp_domain.STATE_PROPERTY_WRITTEN_TRANSLATIONS),
+                    'state_name': change.state_name,
+                    'new_value': (
+                        state_domain.WrittenTranslations.
+                        convert_html_in_written_translations(
+                            change.new_value,
+                            html_validation_service.
+                            add_math_content_to_math_rte_components))
+                })
+
+            elif (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                  change.property_name ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_DEFAULT_OUTCOME):
+                draft_change_list[i] = exp_domain.ExplorationChange({
+                    'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                    'property_name': (
+                        exp_domain.STATE_PROPERTY_INTERACTION_DEFAULT_OUTCOME),
+                    'state_name': change.state_name,
+                    'new_value': (
+                        state_domain.Outcome.convert_html_in_outcome(
+                            change.new_value,
+                            html_validation_service.
+                            add_math_content_to_math_rte_components))
+                })
+
+            elif (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                  change.property_name ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_HINTS):
+                draft_change_list[i] = exp_domain.ExplorationChange({
+                    'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                    'property_name': (
+                        exp_domain.STATE_PROPERTY_INTERACTION_HINTS),
+                    'state_name': change.state_name,
+                    'new_value': (
+                        [(
+                            state_domain.Hint.convert_html_in_hint(
+                                hint, html_validation_service.
+                                add_math_content_to_math_rte_components))
+                         for hint in change.new_value])
+                })
+
+            elif (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                  change.property_name ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_SOLUTION):
+                change.new_value['explanation']['html'] = (
```

Comment:
Why not do it the way its done for hints?

Reply:
done